components/brave_wallet/browser/json_rpc_response_parser.cc

```diff
@@ -292,4 +303,73 @@ bool ParseUnstoppableDomainsProxyReaderGet(const std::string& json,
   return brave_wallet::DecodeString(offset, result, value);
 }
 
+}  // namespace eth
+
+namespace solana {
+
+void ParseErrorResult(const std::string& json,
+                      mojom::SolanaProviderError* error,
+                      std::string* error_message) {
+  int error_code = 0;
+  ::ParseErrorResult(json, &error_code, error_message);
+  *error = static_cast<mojom::SolanaProviderError>(error_code);
+}
+
+bool ParseGetBalance(const std::string& json, uint64_t* balance) {
+  DCHECK(balance);
+
+  base::Value result;
+  if (!ParseResult(json, &result) || !result.is_dict())
+    return false;
+
+  // uint64
+  const base::Value* value = result.FindKey("value");
+  if (!value)
+    return false;
+
+  if (value->is_int() || value->is_double()) {
+    std::string balance_string = base::NumberToString(value->GetDouble());
+    return base::StringToUint64(balance_string, balance);
```

Comment:
Could you post a follow up here for `uint64_t` usage.
I don't think we need to worry about it today but
max value `2^64` = `1.844674407370955e19 - 1`
circulating supply is `314,010,465` and a lamport is `0.000000001`
So that's e18 which would fit, but I believe the supply is not fixed and has no cap so it could be exceeded one day.
Let me know if something looks off above with my reasoning.

Reply:
I see the documentation suggests u64 here https://docs.solana.com/developing/clients/jsonrpc-api#getbalance so I guess it's ok without posting an issue. I assume they'll ensure the supply never goes beyond a factor of 10 higher.